prowler/providers/aws/services/ec2/ec2_instance_port_sqlserver_exposed_to_internet/ec2_instance_port_sqlserver_exposed_to_internet.py

```diff
@@ -0,0 +1,43 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.ec2.ec2_client import ec2_client
+from prowler.providers.aws.services.ec2.lib.security_groups import check_security_group
+from prowler.providers.aws.services.vpc.vpc_client import vpc_client
+
+
+class ec2_instance_port_sqlserver_exposed_to_internet(Check):
+    # EC2 Instances with SQL Server ports open to the Internet will be flagged as FAIL with a severity of medium if the instance has no public IP, high if the instance has a public IP but is in a private subnet, and critical if the instance has a public IP and is in a public subnet.
+    def execute(self):
+        findings = []
+        check_ports = [1433, 1434]
+        for instance in ec2_client.instances:
+            report = Check_Report_AWS(self.metadata())
+            report.region = instance.region
+            report.status = "PASS"
+            report.status_extended = f"Instance {instance.id} does not have SQL Server ports open to the Internet."
+            report.resource_id = instance.id
+            report.resource_arn = instance.arn
+            report.resource_tags = instance.tags
+            if instance.security_groups:
+                for instance_sg in instance.security_groups:
+                    for sg in ec2_client.security_groups:
+                        if sg.id == instance_sg["GroupId"]:
+                            for ingress_rule in sg.ingress_rules:
+                                if check_security_group(
+                                    ingress_rule, "tcp", check_ports, any_address=True
+                                ):
+                                    # The port is open, now check if the instance is in a public subnet with a public IP
+                                    report.status = "FAIL"
+                                    if instance.public_ip:
+                                        report.status_extended = f"Instance {instance.id} has SQL Server exposed to 0.0.0.0/0 on public ip address {instance.public_ip} but in private subnet {instance.subnet_id}."
+                                        report.check_metadata.Severity = "high"
+                                        if vpc_client.vpc_subnets[
+                                            instance.subnet_id
+                                        ].public:
+                                            report.status_extended = f"Instance {instance.id} has SQL Server exposed to 0.0.0.0/0 on public ip address {instance.public_ip} in public subnet {instance.subnet_id}."
+                                            report.check_metadata.Severity = "critical"
+                                    else:
+                                        report.status_extended = f"Instance {instance.id} has SQL Server exposed to 0.0.0.0/0 but with no public ip address."
+                                        report.check_metadata.Severity = "medium"
+                                    break
```

Comment:
Can you think on a way to have this just in one place? All the checks have the same logic.

Reply:
I'm not sure it is that important to say "SQL Server", which means you could have the status extended just use {port} instead.

Reply:
I think the same, at the end any port can be bound to whatever service.

Reply:
Okay, sounds good.

Reply:
@jchrisfarris we will need to change the `check_security_group` function to return the ports that are open. I have included that task in our roadmap, until then, we will leave it as it is now saying the specific service that is open.